### ISSUE TITLE
Need To Separate Trigger Mechanism To Create NAV SO/PO records

### ISSUE THREAD

**Issue by nmfujimoto** [meridianww/M4PLBase#316]
For billing purposes we need to separate when a SO and PO are created in NAV by M4PL. The current behavior is that once a job is delivered AND the POD is uploaded then the SO and PO for the job are created in NAV. 

New behavior:
When the job is delivered (Delivered gateway) create the SO in NAV. 
When the job has the POD (POD Completion gateway) create the PO in NAV.

**Comment by prashantaggarwal1999** [comment]
Hi @nmfujimoto, currently SO/PO generation is happening either manually by clicking on the Button Create/Update Order or automatically while adding the POD Completion Gateway. Changes have been done on a separate branch for Gateway, now if Delivered Gateway added then SO will create and when POD Completion added then PO will create, there are a couple of queries from our side here.

Currently SO/PO creation process is not starting until Job does not complete, and it should not have a valid Origin and Delivery DateTime actuals. Do we need to keep this logic as it is?
Do we need two options on the UI now one for SO creation and one for PO Creation?

**Comment by nmfujimoto** [comment]
> Currently SO/PO creation process is not starting until Job does not complete, and it should not have a valid Origin and Delivery DateTime actuals. Do we need to keep this logic as it is?
Yes, we should keep this logic. The Origin and Delivery DateTime Actual values should be populated either before or at the same time as the Delivered gateway is added to the job. The only difference here is the SO will be created when the order is delivered (DateTime Actual populated/Delivered gateway) and the PO will be created when the POD gateway is added. 

> Do we need two options on the UI now one for SO creation and one for PO Creation?
We might want to consider a setting for when SO/PO creation is performed and link the creation to a specific gateway record. Simplest solution is a checkbox to create SO or PO. Another option is a "Create" drop-down list: "SO/PO, SO, PO". Either approach ties a gateway to the SO or PO creation based on a gateway being added to a job.
